src/asr_got_reimagined/domain/stages/stage_2_decomposition.py

```
@@ -16,167 +16,233 @@
+                if isinstance(meta_val, datetime): props[f"metadata_{meta_field}"] = meta_val.isoformat()
+                elif isinstance(meta_val, Enum): props[f"metadata_{meta_field}"] = meta_val.value
+                # Simplified: assume edge metadata fields are simple or JSON serializable strings
+                elif isinstance(meta_val, (list,set,dict)) or hasattr(meta_val, 'model_dump'):
+                    try: props[f"metadata_{meta_field}_json"] = json.dumps(meta_val.model_dump() if hasattr(meta_val, 'model_dump') else meta_val)
+                    except TypeError: props[f"metadata_{meta_field}_str"] = str(meta_val)
+                else: props[f"metadata_{meta_field}"] = meta_val
+        return {k: v for k, v in props.items() if v is not None}
+
+
+    def _get_conceptual_dimensions(
+        self, 
```

Ruff (0.11.9)

Trailing whitespace

Remove trailing whitespace

(W291)